pyproject.toml

```diff
@@ -27,46 +27,53 @@ module = [
   "_pyodide_core",
   "docutils.parsers.rst",
   "js",
-  "pyodide_js",
-  "ruamel.yaml",
+  "loky",
   "matplotlib.*",
   "PIL.*",
-  "virtualenv",
+  "pyodide_js",
+  "pytest_pyodide",
+  "ruamel.yaml",
   "termcolor",
   "test",
-  "tomllib",
   "tomli",
+  "tomllib",
   "typer",
-  "pytest_pyodide",
-  "loky",
+  "virtualenv",
 ]
 ignore_missing_imports = true
 
 [tool.ruff]
 select = [
-  "E",     # pycodestyles
-  "W",     # pycodestyles
-  "F",     # pyflakes
   "B0",    # bugbear (all B0* checks enabled by default)
   "B904",  # bugbear (Within an except clause, raise exceptions with raise ... from err)
   "B905",  # bugbear (zip() without an explicit strict= parameter set.)
-  "UP",    # pyupgrade
+  "C9",    # mccabe complexity
+  "E",     # pycodestyles
+  "W",     # pycodestyles
+  "F",     # pyflakes
   "I",     # isort
   "PGH",   # pygrep-hooks
+  "PLC",   # pylint conventions
+  "PLE",   # pylint errors
+  "UP",    # pyupgrade
 ]
-# Remove E999 once pattern matching is supported
-# https://github.com/charliermarsh/ruff/issues/282
-ignore = ["E402", "E501", "E731", "E741", "E999"]
+ignore = ["E402", "E731", "E741"]
+line-length = 264
```

Comment:
Why is this needed? If there is a very long line somewhere I would have though it should be skipped explicitly. 
The line length we have should be still around 88 as black.

Reply:
% `ruff --select=E501 --line-length=88 --statistics .`
```
260	E501	Line too long (104 > 88 characters)
```
Leaving `line-length` explicit encourages contributors not to add even longer lines.  Using `--ignore = ["E501"]` puts no such guardrails in place.

If there were a dozen long lines then I would have just shortened them.  It is kinda ugly to add `  # noqa: E501` in the codebase 260 times.  It would be a great `first-issue` to reduce long lines by 10 or 20 characters per line.

Reply:
OK, then I would have preferred to keep the E501 ignore. Most of the code is going to be reformatted by black or prettier anyway, as the value of 264 seems rather arbitrary and maybe a bit misleading with respect to the line length we do expect in the code.

Reply:
[The repo could use more ![good_first_issue](https://user-images.githubusercontent.com/3709715/224486099-b32dc28c-feff-43f1-9bdc-5de59616c400.png)](https://github.com/pyodide/pyodide/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) ;-)

Reply:
long-lines = ∞